### PR TITLE
Select items components

### PR DESCRIPTION
**Pull request by bagby**
Finishes #15 

The `choose-one` and `choose-many` components could probably be refactored into one `choose-items` component, other than the fact that they use radio buttons and checkboxes, respectively, they are nearly identical.

### CODE REVIEW

**Comment by ef4** [comment]
GitHub has [draft PRs](https://github.blog/2019-02-14-introducing-draft-pull-requests/) as a real feature now, so we don't need to keep writing "WIP, don't merge".

**Comment by bagby** [comment]
@ef4 yeah I remembered that about 0.2 seconds after I clicked "Create Pull Request" 🤦‍♂

File: addon/components/choose-many.js [hunk 1]
```diff
@@ -0,0 +1,17 @@
+import Component from '@ember/component';
+import layout from '../templates/components/choose-many';
+
+export default Component.extend({
+  layout,
+  classNames: ['cs-component--choose-many'],
```
Comment (burieberry) [inline addon/components/choose-many.js]: I think we can just name the isolated block parent component `cs-component-choose-many`, and then only use the `--` prefix for its dependents (ie. `cs-component-choose-many--label`) and modifiers. More BEM-like.

File: addon/styles/components/choose-many.scss [hunk 2]
```diff
@@ -0,0 +1,58 @@
+.cs-component--choose-many {
```
Comment (burieberry) [inline addon/styles/components/choose-many.scss]: I'm still not convinced we have to wrap the whole css file inside the main class. I think it increases the below classes' specificity unnecessarily. There's no point for using prefixed class names if we can't reference them directly. I think we should only allow small modifications and states like `&:hover` `&.checked` to be nested.

File: addon/styles/components/choose-many.scss [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+
+    &:hover {
+      background-color: $light-gray;
+    }
+
+    &.checked {
+      background-color: $gray;
+      color: white;
+
+      .cs-component--choose-many--name:after {
```
Comment (burieberry) [inline addon/styles/components/choose-many.scss]: Too much nesting? Now we're at `.cs-component--choose-many .cs-component--choose-many--label.checked .cs-component--choose-many--name:after`

**Review by burieberry — COMMENTED** [review]
Left some comments. I think we should pick one: either prefixed class names or nesting. I remember nesting didn't work out so great for us the last time. It's fine for small changes immediately related to a class, though (like `&:hover`).

**Comment by bagby** [comment]
@burieberry flattened the CSS a bit and made the selectors more BEM-y

**Comment by burieberry** [comment]
Looks good 👍

**Pull request merged**